livekit-plugins/livekit-plugins-openai/livekit/plugins/openai/_oai_api.py

```diff
@@ -115,6 +116,16 @@ def type2str(t: type) -> str:
             if arg_info.choices:
                 p["enum"] = arg_info.choices
 
+            if arg_info.type is int and arg_info.choices and model is not None:
+                from .models import VertexModels
+
+                vertex_models_set = set(get_args(VertexModels))
+
+                if model in vertex_models_set:
+                    raise ValueError(
+                        f"Parameter '{arg_info.name}' uses 'choices' with 'int', which is not supported by Vertex AI."
+                    )
+
```

Comment:
One issue I see with this approach is that we have to keep a up to date list of the supported models Google has. Maybe we can simply create a dataclass with supported stuff?

like:

@dataclass
class SupportedFunctionArguments:
 ....

Reply:
That's a good idea, I'll update pr

Reply:
@theomonnom  I've updated code with LLMCapabilities class